src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -195,23 +187,33 @@ public void addMetadata(String datasetId, List<MetadataInfo> metadataInfos) thro
               datasetId,
               Map.of());
           irmStorageAdaptor.updateStatus(datasetId, STORED);
-          indexDatasetClient.indexDataset(
-              datasetId,
-              UriComponentsBuilder.fromUri(storeUrl)
-                  .path(StoreController.RETRIEVE_DATA_URL_TEMPLATE)
-                  .build(datasetId, IRM_TYPE));
           break;
         default:
           throw new IllegalArgumentException(
               String.format("Received unsupported dataType {%s}", dataType));
       }
     }
+
+    unstage(datasetId);
```

Comment:
❓ Why is `unstage` in the StoreService interface if it is inly used in `StoreServiceImpl`?

Reply:
Not sure. I could potentially see it being used later from the controller. We should leave it as is until we re-do these interfaces.